### ISSUE TITLE
P0009 reference, extensibility testing

### ISSUE THREAD

**Issue by hcedwar** [ORNL/cpp-proposals-pub#69]
Reference implementation strengthened with a little extensibility testing in test subdirectory
- [ ]  layout_none, just plug in an identity map that requires (static_assert) extent::rank() <= 1
- [ ]  AccessPolicy::pointer == std::shared_ptr

**Comment by crtrott** [comment]
Yeah agree, we should do that. Or how about an `extensions` directory next to include and test which adds a few of those things instead of having them only in the test directory? I also was thinking of adding an "include-optimized" or so variant which has things like special map operators for rank 0-4 and the markups to allow interoperability with CUDA and ROCm etc. .
